### PR TITLE
Implement Tablet::get_compaction_status for primary key tablet

### LINKED ISSUE #2884: Implement Tablet::get_compaction_status for primary key tablet
Add an HTTP interface to get the compaction status of the primary-key tablet.
```
wget http://be_host:webserver_port/api/compaction/show?tablet_id=xxxx\&schema_hash=yyyy
```

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [x] enhancement

## Which issues of this PR fixes ：

Fixes  https://github.com/StarRocks/starrocks/issues/2884

## Problem Summary(Required) ：

Starrocks provides an HTTP interface to get the compaction status of tablet, but the primary key tablet is not supported so far. 
This pr add implement for primary key tablet.

### CODE REVIEW

**Review by chaoyli — DISMISSED** [review]

File: be/src/storage/tablet_updates.cpp [hunk 1]
```diff
@@ -1526,11 +1526,78 @@ Status TabletUpdates::compaction(MemTracker* mem_tracker) {
 
     Status st = _do_compaction(&info, true);
     if (!st.ok()) {
+        _last_compaction_failure_millis = UnixMillis();
         _compaction_running = false;
     }
+    _last_compaction_success_millis = UnixMillis();
```
Comment (decster) [inline be/src/storage/tablet_updates.cpp:1532]: else {
    _last_compaction_success_millis = UnixMillis();
}

File: be/src/storage/tablet_updates.cpp [hunk 2]
```diff
@@ -1526,11 +1526,78 @@ Status TabletUpdates::compaction(MemTracker* mem_tracker) {
...
+    rowset_version.SetString(version_str.c_str(), version_str.length(), root.GetAllocator());
+    root.AddMember("rowset_version", rowset_version, root.GetAllocator());
+
+    rapidjson::Document rowsets_arr;
+    rowsets_arr.SetArray();
+    for (int i = 0; i < rowset_ids.size(); ++i) {
+        rapidjson::Value value;
+        std::string rowset_str;
+        if (rowsets[i] != nullptr) {
+            rowset_str = strings::Substitute("id:$0 #seg:$1 #OVERLAP:$2", rowset_ids[i], rowsets[i]->num_segments(),
```
Comment (decster) [inline be/src/storage/tablet_updates.cpp:1584]: primary key rowset's segments never overlap, don't need to print.

File: be/src/storage/tablet_updates.cpp (hunk 1 above)
Comment (sevev) [reply be/src/storage/tablet_updates.cpp]: > else { _last_compaction_success_millis = UnixMillis(); }

done, fix in https://github.com/StarRocks/starrocks/pull/3123/commits/b4f15d173b87b827760de418dc60fe7c3b1dc1ec

File: be/src/storage/tablet_updates.cpp (hunk 2 above)
Comment (sevev) [reply be/src/storage/tablet_updates.cpp]: > primary key rowset's segments never overlap, don't need to print.

done, change in https://github.com/StarRocks/starrocks/pull/3123/commits/b4f15d173b87b827760de418dc60fe7c3b1dc1ec

**Review by decster — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Pull request merged**